### PR TITLE
feat: re-organize resources

### PR DESCRIPTION
**Pull request by amritk**
This PR puts all resources on the base `useWorkspace` store. Then all references in objects are just uids and used to grab the resource from the base.

All resources (requests, requestExamples, servers, collections, folders) all have mutators now and createResource functions!

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -97,6 +97,7 @@ module.exports = {
         ],
         '@typescript-eslint/prefer-optional-chain': 'warn',
         'camelcase': 'off',
+        'max-len': ['warn', { code: 120, comments: 120 }],
```
Comment (amritk) [inline .eslintrc.cjs:100]: I added this to know when to wrap my comments, we can keep it or ditch it.

File: packages/client-app/src/views/Request/Request.vue [hunk 2]
```diff
@@ -63,43 +57,46 @@ executeRequestBus.on(async () => {
...
-        folders,
-        children: Object.keys(folders),
-        requests: _requests,
-      } as const
-    })
-  }
-})
+const workspaceCollections = computed(() =>
+  workspace.collectionUids.map((uid) => collections[uid]),
+)
```
Comment (amritk) [inline packages/client-app/src/views/Request/Request.vue:62]: will bring this back when we need it

File: packages/client-app/src/views/Request/Request.vue [hunk 3]
```diff
@@ -115,79 +112,79 @@ const onDragEnd = (
...
-      newChildren.splice(hoveredIndex + offset, 0, draggingUid)
-
-      collectionMutators.edit(
-        draggingCollectionIndex,
-        `folders.${hoveredParentUid}.children`,
-        newChildren,
-      )
-    }
-  }
+  // const { id: draggingUid, parentId: draggingParentUid } = draggingItem
```
Comment (amritk) [inline packages/client-app/src/views/Request/Request.vue:115]: actually much easier to do drag and drop now so will fast follow with it

**Review by marclave — APPROVED** [review]
huge, lets go!

**Pull request merged**